packages/rspack/src/compiler.ts

```diff
@@ -364,6 +365,9 @@ class Compiler {
 		);
 		this.#updateDisabledHooks();
 	}
+	async #optimize_modules() {
+		await this.compilation.hooks.optimizeModules.promise(this.compilation.getModules())
```

Comment:
pelase add `this.#updateDisabledHooks();` after this line.

Reply:
and this hook needs to be added to the hookMap in `#updateDisabledHooks`

Reply:
@nieyuyao do you have time to address this review issue? ask me anything if you meet any problems.

Reply:
Sorry,  I've been a bit busy lately, I will address the issue as soon as possible.